simpleviewinc/keg-test-consumer
# Add rga4

**Ticket**: [ZEN-508](https://jira.simpleviewtools.com/browse/ZEN-508)

[Semver-Status] minor

### Notes
> * I created a new [repo](https://github.com/lancetipton/rga4) to make setting up Analytics simple
>   * It's a wrapper around google analytics 4
>   * There's also a docs page [here](https://lancetipton.github.io/rga4/)
>   * Feel free to check them out if your interested, and let me know if you have any questions about it

## Context

* Ben needs the technical specifications ready for coding Google Analytics within the X5, so that the sales team can sell the overall product with this included at the end of January.
* The X5 product needs Google Analytics implemented in order to be sold. The functional specifications focus on the main features that need implementing, that convey screen views and in each screen view pass over some metadata e.g person type and event information.
* We want to implement a small number of standard GA "Event" types, but the initial focus is getting all the screen views passing across the page/modal names.


## Goal

* Show an example of google analytics
* Ensure `page_view` events are being captured


## Updates

* `config-overrides.js`
  * Added config for rga4 repo
  * Added webpack override method
* `container/mutagen.yml`
  * Added sync actions for RGA4 repo
* `container/values.yml`
  * Added paths for RGA4 repo
* `package.json`
  * Moved deps into dev
* `src/App.js`
  * Split Sessions component into own file
  * Added RGA4 Provider
    * Uses my own GA ID
* `src/EvfSessions.js`
  * Added code to send google analytics events
    * For booking and for waiting actions
    * Just an example for how to use the RGA4 repo


## Testing

**Setup**
* You should have received an email invite to my demo google analytics account
* Make sure you accept the invite so you can test the analytic events

**Test 1**
* Pull the docker package => `keg consumer pack run docker.pkg.github.com/simpleviewinc/keg-packages/keg-test-consumer:add-rga4`
  * Navigate to http://consumer-add-rga4.local.kegdev.xyz/
  * Ensure the app works as expected
  *  Ensure no errors are thrown

**Test 2**
* Open a new tab and login into the Google Analytics Portal => https://analytics.google.com/analytics/web/
  * Switch the the RGA4-App if not on it already
    * It should look like [this](http://snpy.in/EDq5wX)
* Select the `debug tab` on the left side under the `CONFIGURE` section
   * It should be the last item on the left hand side menu
  * You should now see a timeline of events
  * Looks like [this](http://snpy.in/NxA3aI)
  * You may not see any events when it first loads
* Switch back to the tap-consumer app in your browser
* Select on of the events, and then click `Book Selected` in the modal
  * This should trigger an event in the time-line view of google analytics
* Switch back to the google analytics tab, and ensure the `booking_request` event shows up 
  * Looks like [this](http://snpy.in/0Xw1KI)
  * It may take a couple minutes for the event to show. It says it's real time, but there's sometimes a bit of a delay
* Ensure the booking_request shows up in the google event timeline
